core/cli/src/lib.rs

```diff
@@ -219,160 +313,94 @@ where
 		)
 	}
 
-	let base_path = base_path(&matches);
+	let base_path = base_path(&cli.shared_params);
 
-	config.keystore_path = matches.value_of("keystore")
-		.map(|x| Path::new(x).to_owned())
+	config.keystore_path = cli.keystore_path
 		.unwrap_or_else(|| keystore_path(&base_path, config.chain_spec.id()))
 		.to_string_lossy()
 		.into();
 
-	config.database_path = db_path(&base_path, config.chain_spec.id()).to_string_lossy().into();
-	config.database_cache_size = match matches.value_of("database_cache_size") {
-		Some(s) => Some(s.parse().map_err(|_| "Invalid Database Cache size specified")?),
-		_=> None
-	};
-	config.pruning = match matches.value_of("pruning") {
-		Some("archive") => PruningMode::ArchiveAll,
+	config.database_path =
+		db_path(&base_path, config.chain_spec.id()).to_string_lossy().into();
+	config.database_cache_size = config.database_cache_size;
```

Comment:
This line has no effect, was the assignment maybe intended to be something else?